pkg/ctl/cmdutils/scale.go

```diff
@@ -63,14 +85,11 @@ func validateNameArgument(cmd *Cmd, ng *api.NodeGroupBase) error {
 		ng.Name = cmd.NameArg
 	}
 
-	if ng.Name == "" {
```

Comment:
Huh?

Reply:
Oh, of course, because now we can do this. Right?

Reply:
right! if `scale` is called without `--name`, it scales all nodegroups in the config file, provided there is a config file. 

the name will never be `""` at this point. we only get to this point if either `ng.Name` or `cmd.NameArg` are set, which is checked before the config file loader, [here](https://github.com/weaveworks/eksctl/pull/4464/files/c0fb213ac7f81ae11a0db4acda1bc9f23a6161d1#diff-e9fe30ff1a878b24cb1d115bad2744799bdd46cccbeaee1cdb9c66360262f3a8R59)